client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -816,6 +817,19 @@ static Request verifyRepository(VerifyRepositoryRequest verifyRepositoryRequest)
         return request;
     }
 
+    static Request createSnapshot(CreateSnapshotRequest createSnapshotRequest) throws IOException {
```

Comment:
There are a few extra options i see in the transport layer. Lets make sure we account for all these params / optional things.

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java#L76-L82